src/cls/IPM/Main.cls

```diff
@@ -1871,16 +1942,40 @@ ClassMethod Load(ByRef pCommandInfo) [ Internal ]
 	}
 }
 
+ClassMethod CheckModuleNamespace() As %Status
+{
+	Do ..GetMapInfo(.tIsMappedFrom, .tIsMappedTo)
+	If $Data(tIsMappedFrom($Namespace), sourceNS) # 2 {
+		Quit $$$ERROR($$$GeneralError, $$$FormatText("Cannot install '%1' in namespace '%2' because it is mapped from namespace '%3'.", $$$IPMModuleName, $Namespace, sourceNS))
+	}
+	If $Data(tIsMappedTo($Namespace)) / 2 {
+		Write !, "Will affect the following namespaces:", $NAMESPACE
+		Set ns = ""
+		For {
+			Set ns = $Order(tIsMappedTo($Namespace, ns))
+			If ns = "" {
+				Quit
+			}
+			Write ", ", ns
+		}
+	}
+	Quit $$$OK
+}
+
 ClassMethod Install(ByRef pCommandInfo) [ Internal ]
 {
 	Set tRegistry = ""
 	Set tModuleName = $Get(pCommandInfo("parameters","module"))
 	If (tModuleName["/") {
 		Set $ListBuild(tRegistry, tModuleName) = $ListFromString(tModuleName, "/")
 	}
-  If (tModuleName = "") {
-    Quit $$$OK
-  }
+	If (tModuleName = "") {
+		Quit $$$OK
+	}
+
+	If tModuleName = $$$IPMModuleName {
```

Comment:
I think this check should be moved to LoadNewModule instead to catch the case from the load command as well instead of just the install command. Before loading the file using $System.OBJ.Load(), can read the module name using xslts to confirm whether it can be loaded in the current namespace.